File: sqlite-android-studio-plugin/src/main/java/com.alecstrong.sqlite.android/psi/TableNameElementRef.java
package com.alecstrong.sqlite.android.psi;

import com.alecstrong.sqlite.android.SQLiteParser;
import com.alecstrong.sqlite.android.lang.SqliteTokenTypes;
import com.intellij.psi.tree.IElementType;

public class TableNameElementRef extends SqliteElementRef {
  public TableNameElementRef(IdentifierElement idNode, String ruleName) {
    super(idNode, ruleName);
  }

  @Override public Class<? extends SqliteElement> identifierParentClass() {
    return TableNameElement.class;
  }

  @Override public IElementType identifierDefinitionRule() {
    return SqliteTokenTypes.RULE_ELEMENT_TYPES.get(SQLiteParser.RULE_create_table_stmt);
  }
}

File: sqlite-android-studio-plugin/src/main/kotlin/com/alecstrong/sqlite/android/psi/TableNameElementRef.kt
package com.alecstrong.sqlite.android.psi

import com.alecstrong.sqlite.android.SQLiteParser.RULE_create_table_stmt
import com.alecstrong.sqlite.android.lang.SqliteTokenTypes.RULE_ELEMENT_TYPES
import com.alecstrong.sqlite.android.psi.SqliteElement.TableNameElement

internal class TableNameElementRef(idNode: IdentifierElement, ruleName: String)
: SqliteElementRef(idNode, ruleName) {
  override protected val identifierDefinitionRule = RULE_ELEMENT_TYPES[RULE_create_table_stmt]
}
